src/goto-harness/function_call_harness_generator.cpp

```diff
@@ -278,3 +358,74 @@ void function_call_harness_generatort::implt::
     function_to_call.mode);
   body.add(goto_programt::make_end_function());
 }
+
+code_function_callt::argumentst
+function_call_harness_generatort::implt::declare_arguments(
+  code_blockt &function_body)
+{
+  const auto &function_to_call = lookup_function_to_call();
+  const auto &function_type = to_code_type(function_to_call.type);
+  const auto &parameters = function_type.parameters();
+
+  code_function_callt::operandst arguments{};
+
+  auto allocate_objects = allocate_objectst{function_to_call.mode,
+                                            function_to_call.location,
+                                            "__goto_harness",
+                                            *symbol_table};
+  std::map<irep_idt, irep_idt> parameter_name_to_argument_name;
+  for(const auto &parameter : parameters)
+  {
+    auto argument = allocate_objects.allocate_automatic_local_object(
+      parameter.type(), parameter.get_base_name());
+    parameter_name_to_argument_name.insert(
+      {parameter.get_base_name(), argument.get_identifier()});
+    arguments.push_back(argument);
+  }
+
+  for(const auto &p : parameter_name_to_argument_name)
```

Comment:
⛏ Why not call it a `pair`? it wouldn't cause line wrap. Not using abbreviations is in our coding standard after all.